ols/src/query_helpers/__init__.py

```diff
@@ -1,16 +1,28 @@
 """Question validators, statament classifiers, and response generators."""
 
+import logging
 from typing import Optional
 
+from ols.src.llms.llm_loader import LLMLoader
 from ols.utils import config
 
+logger = logging.getLogger(__name__)
+
 
 class QueryHelper:
     """Base class for query helpers."""
 
     def __init__(
-        self, provider: Optional[str] = None, model: Optional[str] = None
+        self,
+        provider: Optional[str] = None,
+        model: Optional[str] = None,
+        llm_loader: Optional[LLMLoader] = None,
     ) -> None:
         """Initialize query helper."""
+        # NOTE: As signature of this method is evaluated before the config,
+        # is loaded, we cannot use the config directly as defaults and we
+        # need to use those values in the init evaluation.
         self.provider = provider or config.ols_config.default_provider
         self.model = model or config.ols_config.default_model
+        self.llm_loader = llm_loader or LLMLoader
+        self.llm_params: dict = {}
```

Comment:
why not make this an optional parameter of the init function?  I see later you added it as a parameter of the derived classes, but then each derived class is responsible for setting this value in its init function.

Reply:
changed